LEGO1/omni/src/stream/mxramstreamprovider.cpp

```diff
@@ -109,10 +110,8 @@ MxU32 ReadData(MxU8* p_buffer, MxU32 p_size)
 	MxU8* data = p_buffer;
 	MxU8* data2;
 
-#define IntoType(p) ((MxU32*) (p))
-
 	while (data < p_buffer + p_size) {
-		if (*IntoType(data) == FOURCC('M', 'x', 'O', 'b')) {
+		if (data + sizeof(MxU32) <= p_buffer + p_size && UnalignedRead<MxU32>(data) == FOURCC('M', 'x', 'O', 'b')) {
```

Comment:
What is this extra check for?

Reply:
To make sure that we can read 4 byters from the buffer, sometimes there is less then 4 bytes left and it will cause an illegal read.

Reply:
Would we get a *SAN error without this check?

Reply:
Yes. The fix to the UB issue causes and A error, both are fixed with this.

Reply:
This should also work:
```cpp
	while (data + sizeof(MxU32) <= p_buffer + p_size) {
		if (UnalignedRead<MxU32>(data) == FOURCC('M', 'x', 'O', 'b')) {
			// stuff
		}
		else {
			data++;
		}
	}
```

Reply:
yes got let's keep the current solution thank you